### PR TITLE
Move reference creation out of nodeMap

### PR DESCRIPTION
**Pull request by jzaffiro**
Tests not passing: 
- obliterate.reconnect.spec.ts: "deletes concurrently inserted segment between group ops"

### CODE REVIEW

File: packages/dds/merge-tree/src/mergeTree.ts [hunk 1]
```diff
@@ -1928,6 +1928,41 @@ export class MergeTree {
 			localSeq,
 			segmentGroup: undefined,
 		};
+		const normalizedStartPos = startPos === "start" || startPos === undefined ? 0 : startPos;
+		const normalizedEndPos =
+			endPos === "end" || endPos === undefined
+				? this.root.mergeTree?.getLength(refSeq, clientId) ?? 0
```
Comment (anthony-murphy) [inline packages/dds/merge-tree/src/mergeTree.ts]:
```suggestion
				? this.getLength(refSeq, clientId)
```

File: packages/dds/merge-tree/src/mergeTree.ts [hunk 2]
```diff
@@ -1928,6 +1928,41 @@ export class MergeTree {
...
+				: endPos;
+
+		const { segment: startSeg } = this.getContainingSegment(
+			normalizedStartPos,
+			refSeq,
+			clientId,
+		);
+		const { segment: endSeg } = this.getContainingSegment(
+			normalizedEndPos - 1,
```
Comment (anthony-murphy) [inline packages/dds/merge-tree/src/mergeTree.ts:1943]: maybe add an assert that normalizedEndPos  is greater than zero

**Comment by anthony-murphy** [comment]
The test that is failing passed previously, so we should keep it working, as something must be different with this implementation.

File: packages/dds/merge-tree/src/mergeTree.ts [hunk 3]
```diff
@@ -1928,6 +1928,41 @@ export class MergeTree {
 			localSeq,
 			segmentGroup: undefined,
 		};
+		const normalizedStartPos = startPos === "start" || startPos === undefined ? 0 : startPos;
```
Comment (titrindl) [inline packages/dds/merge-tree/src/mergeTree.ts:1931]: Consider the empty obliteration cases too. What's preventing `startPos === 'end'`?

Comment (titrindl) [reply packages/dds/merge-tree/src/mergeTree.ts:1931]: I added the following function on my branch for this purpose:
```typescript
/**
 * Returns the given place in InteriorSequencePlace form.
 */
export function normalizePlace(place: SequencePlace): InteriorSequencePlace {
	if (typeof place === "number") {
		return { pos: place, side: Side.Before };
	}
	if (place === "start") {
		return { pos: -1, side: Side.After };
	}
	if (place === "end") {
		return { pos: -1, side: Side.Before };
	}
	return place;
}
```

Comment (titrindl) [reply packages/dds/merge-tree/src/mergeTree.ts:1931]: With an `InteriorSequencePlace` you can do the following:
```typescript
const startPlace = normalizePlace(start);
const startSegment = this.getContainingSegment(startPlace.pos, refSeq, clientId).segment;
obliterate.start = this.createLocalReferencePosition(
	startSegment,
	startPlace.side === Side.Before ? 0 : startSegment?.cachedLength - 1,
	ReferenceType.StayOnRemove | ReferenceType.RangeBegin,
	{ obliterate },
);
```

**Review by anthony-murphy — APPROVED** [review]

**Pull request merged**